response.go

```diff
@@ -144,10 +144,16 @@ func (r *Response) WriteHeader(code int) {
 	r.headersChan <- struct{}{}
 }
 
-func shouldUseCache(req *http.Request) bool {
+func shouldUseCache(req *http.Request, config *Config) bool {
 
-	if req.Method != "GET" && req.Method != "HEAD" {
-		// Only cache Get and head request
+	matchMethod := false
```

Comment:
I think this can be shortened like the following.

```go
func shouldUseCache(req *http.Request, config *Config) cool {
    for _, method := range config.MatchMethods {
        if method == req.Method {
            return true
        }
    }

    return false
}
```

Reply:
Please correct me if I'm wrong but I thought about your proposal and I think it doesn't work. The function breaks with false if a test fails but returns true if and only if all tests pass at the end of the function. That means returning true in the loop is not correct.

Consider a websocket connection that should return false. This connection uses the HTTP method GET. The loop would let the function return true but should return false later in the “is websocket” if block.

Does it make sense?

Reply:
Oops, my bad! I should open the repo to read the whole codebase first. I review these changes from the github website. I didn't notice the code below the changes.

After reading the whole function, I think you're right :))

Thanks!